.github/actions/validate-access/action.yaml

```diff
@@ -0,0 +1,53 @@
+name: Validate Access
+description: 'Check if the workflow is triggered by an admin user'
+
+# This callable workflow checks if the workflow is triggered by
+# a code owner or an image maintainer by testing the github.actor
+# variable against the CODEOWNERS file and the contacts.yaml file
+# under oci/* path
+
+inputs:
+  admin-only:
+    description: 'The protected workflow should only be triggered as a code owner or an image maintainer'
+    required: true
+    default: 'false'
+  image-path:
+    description: 'The path to the image to be built'
+    required: true
+  github-token:
+    description: 'The GH_TOKEN for the GitHub CLI'
+    required: true
+
+runs:
+  using: "composite" 
+  steps:
+    - name: Check if the workflow is triggered by an admin user
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+      run: |
+        echo "github.actor: ${{ github.actor }}"
+        echo "admin-only: ${{ inputs.admin-only }}"
+        if [[ ${{ inputs.admin-only }} == true ]]; then
+          echo "Expanding team mentions in the CODEOWNERS file"
+          teams=$(grep -oE '@[[:alnum:]_.-]+\/[[:alnum:]_.-]+' ${{ github.workspace }}/CODEOWNERS || true | sort | uniq)
+
+          for team in $teams; do
+            org=$(echo $team | cut -d'/' -f1 | sed 's/@//')
+            team_name=$(echo $team | cut -d'/' -f2)
+            members=$(gh api "/orgs/$org/teams/$team_name/members" | jq -r '.[].login')
+            replacement=$(echo "$members" | xargs -I {} echo -n "@{} " | awk '{$1=$1};1')
+            sed -i "s|$team|$replacement|g" ${{ github.workspace }}/CODEOWNERS
+          done
+
+          if grep -wq "@${{ github.actor }}" ${{ github.workspace }}/CODEOWNERS; then
+            echo "The workflow is triggered by ${{ github.actor }} as the code owner"
+          elif cat ${{ github.workspace }}/${{ inputs.image-path }}/contacts.yaml | yq ".maintainers" | grep "\- " | grep -wq "${{ github.actor }}"; then
```

Comment:
```suggestion
          elif cat "${{ github.workspace }}/${{ inputs.image-path }}/contacts.yaml" | yq ".maintainers" | grep "\- " | grep -wq "${{ github.actor }}"; then
```

Reply:
The `${{ github.workspace }}` and `${{ inputs.image-path }}` is not the shell syntax, but GitHub Workflow expression syntax instead. Since we are not passing any white spaces in these variables, we don't have to quote them.